Realm namespace

### What happened?

Since upgrading to version 11.1.0 I get the following compiler error:

```
error CS0118: 'Realm' is a namespace but is used like a type
```
When trying to use the Realm type like this:

```cs
using Realms;

public class Example
{
    private Realm realm;
}
```

Fully qualifying the type does resolve the error:

```cs
public class Example
{
    private Realms.Realm realm;
}
```

 I think this might have something to do with the addition of the Realm namespace in  #3323 

### Repro steps

1. Create a new Unity project and add the realm package
2. Create the following class:

```cs
using Realms;

public class Example
{
    private Realm realm;
}
```

### Version

11.1.0

### What Atlas Services are you using?

Local Database only

### What type of application is this?

Unity

### Client OS and version

Windows 11

### Code snippets

```cs
using Realms;

public class Example
{
    private Realm realm;
}
```

### Stacktrace of the exception/crash you're getting

_No response_

### Relevant log output

```shell
Assets\Example.cs(5,13): error CS0118: 'Realm' is a namespace but is used like a type
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,13 +4,13 @@
 * None
 
 ### Fixed
-* None
+* Fixed a namespacing issue that would cause Maui Android projects to fail to build due to `'Realm' is a namespace but is used like a type`. (Issue [#3351](https://github.com/realm/realm-dotnet/issues/3351))
 
 ### Compatibility
 * Realm Studio: 13.0.0 or later.
 
 ### Internal
-* Using Core x.y.z.
+* Using Core 13.15.0
 
 ## 11.1.1 (2023-06-19)
 

--- a/Realm/Realm.PlatformHelpers/Realm.PlatformHelpers.csproj
+++ b/Realm/Realm.PlatformHelpers/Realm.PlatformHelpers.csproj
@@ -21,6 +21,7 @@
     <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
     <LangVersion>9.0</LangVersion>
     <nullable>enable</nullable>
+    <AndroidResgenNamespace>Realms.PlatformHelpers</AndroidResgenNamespace>
   </PropertyGroup>
 
   <ItemGroup>

